test/unit/sdam/server_selection/select_servers_tests.js

```diff
@@ -0,0 +1,89 @@
+'use strict';
+const ReadPreference = require('../../../../lib/core/topologies/read_preference');
+const Topology = require('../../../../lib/core/sdam/topology');
+const Server = require('../../../../lib/core/sdam/server');
+const serverSelection = require('../../../../lib/core/sdam/server_selection');
+const selectServers = serverSelection.selectServers;
+const sc = require('../../../../lib/core/sdam/common');
+const expect = require('chai').expect;
+const sinon = require('sinon');
+
+describe('selectServers', function() {
+  beforeEach(function() {
+    this.sinon = sinon.sandbox.create();
+  });
+
+  afterEach(function() {
+    this.sinon.restore();
+  });
+
+  it('should error immediately if timeout exceeds start time', function(done) {
+    const topology = new Topology('invalid:27019');
+    const start = process.hrtime();
+    start[0] = start[0] - 500;
```

Comment:
Are we expecting a 500s timeout?

Reply:
yes, see the call to `selectServers` below. The third parameter is the timeout